Firmware/HomeButtonsArduino/src/hardware.cpp

```diff
@@ -154,29 +181,33 @@ void HardwareDefinition::blink_led(uint8_t led, uint8_t num_blinks,
 }
 
 float HardwareDefinition::read_battery_voltage() {
-  return analogRead(VBAT_ADC) / 4095.0 * BATT_ADC_REF_VOLT / BATT_DIVIDER;
+  return (analogReadMilliVolts(VBAT_ADC) / 1000.) / BATT_DIVIDER;
 }
 
 uint8_t HardwareDefinition::read_battery_percent() {
   if (!is_battery_present()) return 0;
-  float pct = 100 * (read_battery_voltage() - BATT_EMPTY_VOLT) /
-              (BATT_FULL_VOLT - BATT_EMPTY_VOLT);
-  if (pct < 0.0)
-    pct = 0;
+  float pct = BATT_SOC_EST_K * read_battery_voltage() + BATT_SOC_EST_N;
+  if (pct < 1.0)
+    pct = 1;
   else if (pct > 100.0)
     pct = 100;
   return (uint8_t)round(pct);
 }
 
-void HardwareDefinition::read_temp_hmd(float &temp, float &hmd) {
+void HardwareDefinition::read_temp_hmd(float &temp, float &hmd,
+                                       const bool fahrenheit) {
   shtc3_wire.begin(
       (int)SDA,
       (int)SCL);  // must be cast to int otherwise wrong begin() is called
   shtc3.begin(&shtc3_wire);
   sensors_event_t humidity_event, temp_event;
   shtc3.getEvent(&humidity_event, &temp_event);
   shtc3.sleep(true);
-  temp = temp_event.temperature;
+  if (fahrenheit) {
+    temp = temp_event.temperature * 1.8 + 32;
+  } else {
+    temp = temp_event.temperature;
+  }
```

Comment:
Disagree. Should be published to MQTT also, since some users are not using Home Assistant. Home Assistant can handle receiving F or C and displays it in the units set by the user.